### PR TITLE
nrf_wifi: SnS Fixes

### PR DESCRIPTION
**Pull request by krish2718**
Fixes for various issues during SnS tests, mainly around recovering nRF70 without host reboot.

### CODE REVIEW

**Review by sachinthegreen — DISMISSED** [review]

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 1]
```diff
@@ -662,6 +663,66 @@ config NRF_WIFI_RPU_RECOVERY_PROPAGATION_DELAY_MS
 config NET_MGMT_EVENT_QUEUE_SIZE
 	# Doing interface down and up even with delay puts a lot of events in the queue
 	default 16
+
+config NRF_WIFI_RPU_RECOVERY_PROPAGATION_DELAY_MS
+	int "RPU recovery propagation delay in milliseconds"
+	default 10
+	help
+		Propagation delay in milliseconds to wait after RPU is powered down
```
Comment (rlubos) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:671]: nit: Help indentation is inconsistent, typically it's just two extra spaces like below.

**Review by rlubos — DISMISSED** [review]

File: drivers/wifi/nrfwifi/src/shim.c [hunk 2]
```diff
@@ -235,6 +235,9 @@ static void *zep_shim_nbuf_alloc(unsigned int size)
 
 static void zep_shim_nbuf_free(void *nbuf)
 {
+	if (!nbuf)
+		return;
```
Comment (jukkar) [inline drivers/wifi/nrfwifi/src/shim.c:241]: `{ }` missing

File: drivers/wifi/nrfwifi/src/fmac_main.c [hunk 3]
```diff
@@ -252,8 +252,8 @@ static void nrf_wifi_process_rssi_from_rx(void *vif_ctx,
 
 	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
 
-	if (!rpu_ctx_zep) {
-		LOG_ERR("%s: rpu_ctx_zep is NULL", __func__);
+	if (!rpu_ctx_zep || !rpu_ctx_zep->rpu_ctx) {
```
Comment (jukkar) [inline drivers/wifi/nrfwifi/src/fmac_main.c:255]: At least for me, I need to twist my brain when seeing multiple negations. We could write this as
`if (!(rpu_ctx_zep && rpu_ctx_zep->rpu_ctx)) {`
which is slightly more readable.

File: drivers/wifi/nrfwifi/src/fmac_main.c [hunk 4]
```diff
@@ -252,8 +252,8 @@ static void nrf_wifi_process_rssi_from_rx(void *vif_ctx,
...
+		LOG_ERR("%s: rpu_ctx_zep is NULL\n", __func__);
```
Comment (jukkar) [inline drivers/wifi/nrfwifi/src/fmac_main.c:256]: Please do not put `\n` to log prints as they cause empty lines to be printed (double newlines)

File: drivers/wifi/nrfwifi/src/shim.c [hunk 5]
```diff
@@ -748,6 +772,11 @@ static void zep_shim_irq_handler(const struct device *dev, struct gpio_callback
 	ARG_UNUSED(cb);
 	ARG_UNUSED(pins);
 
+	if (!intr_priv || !intr_priv->callbk_fn || !intr_priv->callbk_data) {
```
Comment (jukkar) [inline drivers/wifi/nrfwifi/src/shim.c:775]: Similar brain twister here, could be simplified

File: include/zephyr/net/wifi.h [hunk 6]
```diff
@@ -465,6 +467,24 @@ enum wifi_ps_wakeup_mode {
...
+enum wifi_ps_exit_strategy {
+	/** PS-Poll frame based */
+	WIFI_PS_EXIT_CUSTOM_ALGO = 0,
+	/** QoS NULL frame based */
+	WIFI_PS_EXIT_EVERY_TIM,
+
+	/** Last value */
+	WIFI_PS_EXIT_LAST,
+	/** Maximum value */
+	WIFI_PS_EXIT_MAX = WIFI_PS_EXIT_LAST - 1,
```
Comment (jukkar) [inline include/zephyr/net/wifi.h:482]: No need to do add these to documentation, we should surround them with `@cond ... @endcond`

**Review by jukkar — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**